src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -988,28 +1021,76 @@ private async Task SendWithLock(ConnectionState expectedConnectionState, HubMess
         return null;
     }
 
-    private async Task DispatchInvocationAsync(InvocationMessage invocation)
+    private async Task DispatchInvocationAsync(InvocationMessage invocation, ConnectionState connectionState)
     {
+        var expectsResult = !string.IsNullOrEmpty(invocation.InvocationId);
         // Find the handler
         if (!_handlers.TryGetValue(invocation.Target, out var invocationHandlerList))
         {
             Log.MissingHandler(_logger, invocation.Target);
+            if (expectsResult)
+            {
+                Log.NoResultGiven(_logger, invocation.Target, invocation.InvocationId!);
```

Comment:
Can we have different logs for when missing any handler vs having a handler but missing a returning handler?